File: C++/Day 02/day2-part1/day2-part1.cpp
#include "../../AOCHeaders/stdafx.h"


int main()
{
	std::fstream in("input.in", std::fstream::in);
	std::fstream out("output.out", std::fstream::out);

	std::string password;
	int lowest{ 0 }, highest{ 0 }, countPasswords{ 0 }, countLetters{ 0 };
	char letter{}, aux{};

	while (in >> lowest >> aux >> highest >> letter >> aux >> password)
	{
		countLetters = std::count(std::begin(password), std::end(password), letter);

		if ((countLetters >= lowest) && (countLetters <= highest))
		{
			countPasswords++;
		}
	}

	out << countPasswords;

	in.close();
	out.close();
}

File: JavaScript/Day 02/day2-part1/day2-part1.js
const fs = require("fs");
const data = fs
  .readFileSync("input.in", { encoding: "utf8", flag: "r" })
  .toString()
  .split("\n")
  .map((el) => el.split(" "))
  .map((el) => {
    let [lowest, highest] = el[0].split("-");
    return [parseInt(lowest), parseInt(highest), el[1].split(":")[0], el[2]];
  });
let countPasswords = 0;

for (el of data) {
  const [lowest, highest, letter, password] = el;
  let countLetters = 0;

  for (pass of password) {
    if (pass === letter) {
      countLetters++;
    }
  }
  if (countLetters >= lowest && countLetters <= highest) {
    countPasswords++;
  }
}

fs.writeFileSync("output.out", countPasswords.toString());
